entity-framework/efcore-and-ef6/porting/port-behavior.md

```diff
@@ -0,0 +1,49 @@
+---
+title: Porting from EF6 to EF Core - Behavior changes
+description: A non-exhaustive list of some changes in behavior between EF6 and EF Core.
+author: ajcvickers
+ms.date: 10/27/2016
+uid: efcore-and-ef6/porting/port-behavior
+---
+# Behavior Changes between EF6 and EF Core
+
+This is a non-exhaustive list of some changes in behavior between EF6 and EF Core. It is important to keep these in mind as your port your application as they may change the way your application behaves, but will not show up as compilation errors after swapping to EF Core.
+
+## DbSet.Add/Attach and graph behavior
+
+In EF6, calling `DbSet.Add()` on an entity results in a recursive search for all entities referenced in its navigation properties. Any entities that are found, and are not already tracked by the context, are also marked as added. `DbSet.Attach()` behaves the same, except all entities are marked as unchanged.
+
+**EF Core performs a similar recursive search, but with some slightly different rules.**
+
+* The root entity is always in the requested state (added for `DbSet.Add` and unchanged for `DbSet.Attach`).
+* **For entities that are found during the recursive search of navigation properties:**
+  * **If the primary key of the entity is store generated**
+    * If the primary key is not set to a value, the state is set to added. The primary key value is considered "not set" if it is assigned the CLR default value for the property type (for example, `0` for `int`, `null` for `string`, etc.).
```

Comment:
Would be good to link to other docs where they have more information--change tracking in this case.